[Pico 2040] Compilation errors with PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY

### Operating System

Linux

### Arduino IDE version

2.0.4

### Board

Pico 2040

### ArduinoCore version

Arduino Mbed for Pico 2040 version 4.0.2

### TinyUSB Library version

2.0.0

### Sketch as ATTACHED TXT

From the Adafruit TinyUSB library examples:
hid_gamepad.ino

(Or any other)

### Compiled Log as ATTACHED TXT

.../Arduino/libraries/Adafruit_TinyUSB_Library/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp:108:26: error: 'PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY' was not declared in this scope
                          PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY);
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

### What happened ?

Compilation failure.

### How to reproduce ?

Compile any USB example from the 2.0.0 Adafruit library version.

### Debug Log

_No response_

### Screenshots

_No response_

--- a/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp
+++ b/src/arduino/ports/rp2040/Adafruit_TinyUSB_rp2040.cpp
@@ -88,6 +88,10 @@ static void usb_task_irq(void) {
   }
 }
 
+#ifndef PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY
+#define PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY 0x00
+#endif
+
 // invoked when there is hardware usb irq, trigger task runner later
 static void usb_task_trigger_irq(void) { irq_set_pending(USB_TASK_IRQ); }
 

--- a/src/portable/raspberrypi/rp2040/dcd_rp2040.c
+++ b/src/portable/raspberrypi/rp2040/dcd_rp2040.c
@@ -384,6 +384,11 @@ static void __tusb_irq_path_func(dcd_rp2040_irq)(void)
 /* Controller API
  *------------------------------------------------------------------*/
 
+// older SDK
+#ifndef PICO_SHARED_IRQ_HANDLER_HIGHEST_ORDER_PRIORITY
+#define PICO_SHARED_IRQ_HANDLER_HIGHEST_ORDER_PRIORITY 0xff
+#endif
+
 void dcd_init (uint8_t rhport)
 {
   assert(rhport == 0);

